checkov/secrets/plugins/entropy_keyword_combinator.py

```diff
@@ -220,18 +220,39 @@ def analyze_line(
                         kwargs=kwargs
                     )
                 else:
+                    # preprocess line before detecting secrets - add quotes on potential secrets to allow triggering
+                    # entropy detector
                     for pt in keyword_on_key:
                         if pt.secret_value:
                             quoted_secret = f"\"{pt.secret_value}\""
                             if line.find(quoted_secret) < 0:    # replace potential secret with quoted version
                                 line = line.replace(pt.secret_value, f"\"{pt.secret_value}\"", 1)
-                    return self.detect_secret(
+                    detected_secrets = self.detect_secret(
                         scanners=self.high_entropy_scanners_iac,
                         filename=filename,
                         line=line,
                         line_number=line_number,
                         kwargs=kwargs
                     )
+                    # postprocess detected secrets - filter out potential secrets on keyword and re-run secret detection
+                    # on their value only
+                    for detected_secret in detected_secrets:
+                        if detected_secret.secret_value and line.replace('"', '').replace("'", '').startswith(detected_secret.secret_value):
+                            # Found keyword prefix as potential secret
+                            line = line.replace(detected_secret.secret_value, '')
+
+                            # Re-run secret detection on cut line
+                            value_secrets = self.detect_secret(
+                                scanners=self.high_entropy_scanners_iac,
+                                filename=filename,
+                                line=line,
+                                line_number=line_number,
+                                kwargs=kwargs
+                            )
+                            detected_secrets.remove(detected_secret)
+                            detected_secrets = detected_secrets.union(value_secrets)
```

Comment:
```suggestion
                            detected_secrets.update(value_secrets)
```
should be the same without creating a new set 🙂